src/components/search/pages/Search/SearchPage.tsx

```diff
@@ -0,0 +1,173 @@
+
+const mapStateToProps = (state: RootState) => ({
+    result: state.search.result,
+    query: state.search.query,
+});
+
+const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, null, AnyAction>) =>
+    bindActionCreators({ querySearch, openDetail: openDetail }, dispatch);
+
+export type SearchPageReduxStateProps = ReturnType<typeof mapStateToProps>;
+
+export type SearchPageReduxDispatchProps = ReturnType<typeof mapDispatchToProps>;
+
+export type SearchPageReduxProps = SearchPageReduxStateProps & SearchPageReduxDispatchProps;
+
+export type SearchPageProps = SearchPageReduxProps & Themeable;
+
+export class SearchPage extends React.Component<SearchPageProps, {}> {
+    constructor(props: SearchPageProps) {
+        super(props);
+        this.onPaginationClick = this.onPaginationClick.bind(this);
+        this.onDetailClick = this.onDetailClick.bind(this);
+        this.renderResults = this.renderResults.bind(this);
+        this.renderMatchesDesc = this.renderMatchesDesc.bind(this);
+        this.renderSearchElapsedTime = this.renderSearchElapsedTime.bind(this);
+    }
+
+    get pagesCount() {
+        const { result } = this.props;
+        if (result?.data) {
+            return Math.ceil(result.data.total / result.data.limit);
+        }
+        return 0;
+    }
+
+    get currentPage() {
+        const { result } = this.props;
+        if (result?.data) {
+            return result.data.offset / result.data.limit + 1;
+        }
+        return 0;
+    }
+
+    onPaginationClick(pageNum: number) {
+        const { query } = this.props;
+        this.props.querySearch({ ...query, pageNum });
+    }
+
+    onDetailClick(entity: TextItemResponse) {
+        if (entity.name !== undefined && entity.type !== undefined) {
+            this.props.openDetail(stripHtml(entity.name), entity.type);
+        }
+    }
+
+    renderMatchesDesc() {
+        const { theme, result } = this.props;
+        if (!result?.data) {
+            return;
+        }
+        return (
+            <div className={searchPageMatchesDesc(theme)}>
+                <strong data-test="total">{result.data.total}</strong> results
+            </div>
+        );
+    }
+
+    renderSearchElapsedTime() {
+        const { theme, result } = this.props;
+        if (!result?.data) {
+            return;
+        }
+        return (
+            <div className={searchPageElapsed(theme)}>
+                Elapsed:{' '}
+                <strong>
+                    <span data-test="elapsed">{result.data.elapsed}</span>s
+                </strong>
+            </div>
+        );
+    }
+
+    renderResults() {
+        const {
+            props,
+            onPaginationClick,
+            onDetailClick,
+            pagesCount,
+            currentPage,
+            renderMatchesDesc,
+            renderSearchElapsedTime,
+        } = this;
+        const { result, query } = props;
+        if (!result) {
+            return <p>No result data.</p>;
```

Comment:
Wanted to say, that assuming navigation will work OK, never, since SearchPage would not have been navigatable without entering some sort of query. Then I noticed that initial state of "result" does not need to be null, so I will refactor it a bit.

Reply:
Addressed in ba23ab4